### ISSUE TITLE
dataset of pictures

### ISSUE THREAD

**Issue by Azemar** [TheLastBen/fast-stable-diffusion#171]
Is it possible for the training model to specify the path to the dataset of pictures in google drive, instead of downloading them every time from PC?

**Comment by TheLastBen** [comment]
Yes, you can specify the path to the folders in INSTANCE_DIR_OPTIONAL and CLASS_DIR

The script automatically resize the images and crops them if they are not square

**Comment by Azemar** [comment]
> Yes, you can specify the path to the folders in INSTANCE_DIR_OPTIONAL and CLASS_DIR
> 
> The script automatically resize the images and crops them if they are not square

Is there a point in feeding 512x512 images, or are square proportions enough?

**Comment by Azemar** [comment]
I have 250 images of one artist, do I understand correctly that I need to select the top 50 images from them and specify the path to them in the "Setting up" code block?
And the remaining 200 images put in the block "[Optional] Upload or choose a folder of the class pictures"?

Is it possible to enter several values separated by commas in the "SUBJECT_TYPE:" line, for example "sketch, drawing, art"? If so, what's the limit?

In other applications seed was ten digits, but in google colab it was five digits. Does this affect the result and does it make sense to enter 0000045576 instead of 45576?

**Comment by TheLastBen** [comment]
you can enter any number as a seed, no problem

as for the instance images, you can still use all the 250 if you want.

if you choose "artist"or "style" above, the method won't require class images, but if you choose a method that requires class images, you can let the script generate them by skipping the [Optional] cell.

class images provide context to the trained subject, in the case of artist or style, you don't need them.

for subject type, you can use as much words as you want, but keep it narrow because you might confuse the AI

**Comment by Azemar** [comment]
Now there is a new error when using the old method:
---------------------------------------------------------------------------
NameError Traceback (most recent call last)
<ipython-input-22-19dd79e2614d> in <module>
      7 #@markdown ---
      8 
----> 9 if Use_New_Fast_Method=="No":
     10 
     11 Training_Subject = "Artist" #@param ["Character", "Object", "Style", "Artist", "Movie", "TV Show"]

NameError: name 'Use_New_Fast_Method' is not defined

surprisingly, a few hours ago the old method was working fine.  I saw a similar issue and followed the "new link", but it didn't help.

**Comment by TheLastBen** [comment]
you need to set the "Use_New_Fast_Method" to "No" in the new method cell

**Comment by TheLastBen** [comment]
fixed.

**Comment by Azemar** [comment]
Also noticed such a bug or "feature" that if you uncheck the checkbox fp16 in the training settings and select the generation of intermediate models, these models will have a size of 2 GB, ie fp16. The final model will be of normal size - 4gb.

That is, if I generate a model with 7000 steps and also want to get a model with 6000 steps, then the model with 7000 steps will be fp32 and with 6000 steps fp16.

**Comment by TheLastBen** [comment]
I'll see what I can do about that, did you notice a quality difference between fp16 and fp32 ?

**Comment by Azemar** [comment]
> I'll see what I can do about that, did you notice a quality difference between fp16 and fp32 ?

I haven't checked on purpose, but I'll check and write on occasion.

**Comment by Azemar** [comment]
> I'll see what I can do about that, did you notice a quality difference between fp16 and fp32 ?

I compared 20 images on the fp16 and fp32 models, and in most cases the fp32 was better. Regarding generation time - I didn't notice much difference on the 1660TI 6gb. examples are at the link below.
https://imageban.ru/show/2022/11/21/0ba2663d77045acfec3f401ff545415b/png

And in some 25 days everything has changed again, so new questions:
In the "Create/Load a Session" block, you need to rename the name of the images to a unique identifier. I.e. the previous scheme with writing tags for each art in the file name does not work?

In the "Instance Images" block, it is not quite clear what the "Remove_existing_instance_images" checkbox is for. If I understand correctly, then now a permanent session is created and you don't need to specify the path to the image instances every time? If so, how long is this session viable? 

Block "Training" line "Resolution" - if my instances of images resolution 3500x3500, then in this line also need to specify the value of 3500? And if I put less, how much will the quality drop?

The "Enable_text_encoder_training:" line - what does this function do? Automatic recognition of objects in the image, so as not to write tags in the name of the image?
